src/main/java/org/spongepowered/common/event/tracking/TrackingUtil.java

```diff
@@ -792,7 +792,9 @@ private static void spawnEntitiesForBlock(Collection<net.minecraft.entity.Entity
                     }
                 }
                 final ChangeBlockEvent.Post post = ((IPhaseState) state).createChangeBlockPostEvent(context, transactions);
-                SpongeImpl.postEvent(post);
+                if (ShouldFire.CHANGE_BLOCK_EVENT_POST) {
```

Comment:
Only create the event within the ShouldFire

Reply:
Then I would've to return null if the event is not fired.

That would mean that `//Means that we have had no actual block changes apparently?`

https://github.com/SpongePowered/SpongeCommon/blob/stable-7/src/main/java/org/spongepowered/common/event/tracking/TrackingUtil.java#L496-L500